Initial implementation of <MUIPlacesAutocomplete> component

Giners opened the pull request:
This commit is the initial implementation of the <MUIPlacesAutocomplete>
component. The component leverages the <Autosuggest> component from the
'react-autosuggest' package as well as the Google Places API Web
Service/Google Maps JavaScript API to provide autosuggest/completion
functionality when searching for a place. The main feature of the
<MUIPlacesAutocomplete> component is that it provides a Google Material
Design feel to the rendered elements via the Material-UI library.

When reviewing please be critical (in a constructive manner) while taking the viewpoint of someone who is going to download/consume this.

Giners committed:
Initial implementation of <MUIPlacesAutocomplete> component

This commit is the initial implementation of the <MUIPlacesAutocomplete>
component. The component leverages the <Autosuggest> component from the
'react-autosuggest' package as well as the Google Places API Web
Service/Google Maps JavaScript API to provide autosuggest/completion
functionality when searching for a place. The main feature of the
<MUIPlacesAutocomplete> component is that it provides a Google Material
Design feel to the rendered elements via the Material-UI library.

Moving forward I look to iterate on the implementation by querying for
constructive criticisms as to how I can improve my JS/front-end skills
as well as how I can make the component more useful for other consumers.

Included is a demo which people can run locally to see the component in
action.
	modified:   .babelrc
	modified:   .eslintrc
	modified:   .gitignore
	modified:   demo/Demo.jsx
	modified:   demo/index.html
	modified:   package.json
	modified:   src/MUIPlacesAutocomplete.jsx
	new file:   src/images/google-logo-on-white-bg.png
	modified:   src/index.js
	new file:   test/api-key.js
	new file:   test/setup-dom.js
	new file:   test/test.jsx
	new file:   test/test.jsx.snap
	new file:   test/testHelper.js
	new file:   webpack.config.js
	modified:   yarn.lock

Giners committed:
License project under MIT license

While the 'package.json' file already listed the license as the MIT
license this now officially licenses the project with the MIT license.
	new file:   LICENSE.md

Giners commented on src/MUIPlacesAutocomplete.jsx:
```diff
@@ -1,6 +1,213 @@
+  }
+
+  componentWillMount() {
+    this.autocompleteService = new window.google.maps.places.AutocompleteService()
```
Can you give guidance for how to properly consume 3rd party libraries that are expected to be loaded in the browser/window?

For example should I account for cases where the library doesn't load and if so what ought I do? Or is this fine the way it is implemented and if the library doesn't load its okay to let this component blow up and errors show up in the console?

mattbailey replied:
generally, this is hard.  You might want to see if someone has wrapped this kind of thing up before, e.g.: https://www.npmjs.com/package/react-google-maps

Giners replied:
Ya. This seems like a different set of problems that I have never had to consider before.

I saw that package (and others) in the past when I was looking for something myself but thought that they weren't sufficient in terms of getting a MUI feeling applied to them. Maybe I was wrong just based on my limited knowledge.

Not looking to re-invent the wheel but it feels good to produce this myself. I just get to further learn more about the ecosystem Im working with. Plus I have open sourced something! w00t! :)

mattbailey replied:
https://www.npmjs.com/package/google-maps

there ya go, someone did simply wrap it in a module with 0 changes, should check it out

mattbailey replied:
otherwise, yeah, all good

Giners replied:
TY. Im going to make a task to look at this.

mattbailey commented on test/api-key.js:
```diff
@@ -0,0 +1,2 @@
+// Set your Google API key so that it can be used for the tests
+export default 'Replace with your Google API key'
```
better done with env vars, e.g. `process.env.GOOGLE_API_KEY` so you can easily run tests in CI

mattbailey reviewed (commented):
LGTM, just one comment on your api key

Giners committed:
Allow Google API key to be provided as environment var for test

This commit allows the Google API key to be provided as an environment
variable for testing purposes. This will make it easier to perform
testing in a CI/CD pipeline.

This commit is in response to requested changes for PR #2
	modified:   test/api-key.js

Giners committed:
Add README.md

This commit adds the README.md file which includes content for
installing, setting up, using the component, and running a demo of the
component amongst other info.
	new file:   README.md

Giners committed:
Add `contributors` section to `package.json` - Add Matt as contributor

This commit updates the `package.json` file to add a `contributors`
section. We also include Matt Bailey as a contributor to the project.
	modified:   package.json
